esx_service/vmdk_ops.py

```diff
@@ -386,12 +390,19 @@ def parse_vol_name(full_vol_name):
     On parse errors raises ValidationError with syntax explanation
     """
     # note: \w in regexp is [a-zA-Z0-9_]
-    groups = re.match("\A([\w\-.]+)@?([\w\-.]+)?$", full_vol_name)
+    groups = re.match(r"\A([a-zA-Z_][\w\_.]+)(@([a-zA-Z_][\w\_\-.]+))?$", full_vol_name)
     if not groups:
-        raise ValidationError("Invalid syntax: '{0}'. " \
-                           "Valid syntax is volume@datastore, where volume or datastore" \
-                           "can contain [a-zA-Z0-9_-.]".format(full_vol_name))
-    return groups.groups()[0], groups.groups()[1]
+        raise ValidationError("Invalid syntax: '{0}'.\n"
```

Comment:
Numbers should be allowed in the volume name as mentioned in the help message.

>>> s='1@x.y.z'
>>> groups = re.match(r"\A([a-zA-Z_][\w\_.]+)(@([a-zA-Z_][\w\_\-.]+))?$", s)
>>> print(groups)
None
>>> s='1@1'
>>> groups = re.match(r"\A([a-zA-Z_][\w\_.]+)(@([a-zA-Z_][\w\_\-.]+))?$", s)
>>> print(groups)
None